pkg/cmdutil/usage_func.go

```diff
@@ -0,0 +1,226 @@
+	"github.com/spf13/pflag"
+
+	"github.com/algolia/cli/pkg/utils"
+)
+
+func UsageFuncWithFilteredFlags(w io.Writer, command *cobra.Command, flagsToDisplay []string) error {
+	fmt.Fprintf(w, "Usage:  %s", command.UseLine())
+
+	subcommands := command.Commands()
+	if len(subcommands) > 0 {
+		fmt.Fprint(w, "\n\nAvailable commands:\n")
+		for _, c := range subcommands {
+			if c.Hidden {
+				continue
+			}
+			fmt.Fprintf(w, "  %s\n", c.Name())
+		}
+		return nil
+	}
+
+	flagUsages := FlagUsagesFiltered(command.LocalFlags(), flagsToDisplay)
+	if flagUsages != "" {
+		fmt.Fprintln(w, "\n\nFlags:")
+		fmt.Fprintln(w, flagUsages)
+	}
+	return nil
+}
+
+// Customized version of "FlagUsagesWrapped" in github.com/spf13/pflag/flag.go
+// that only display flags passed in arguments.
+func FlagUsagesFiltered(f *pflag.FlagSet, flagsToDisplay []string) string {
+	buf := new(bytes.Buffer)
+
+	lines := []string{}
+
+	maxlen := 0
+	f.VisitAll(func(flag *pflag.Flag) {
+		if flag.Hidden || !utils.Contains(flagsToDisplay, flag.Name) {
+			return
+		}
+
+		line := ""
+		if flag.Shorthand != "" && flag.ShorthandDeprecated == "" {
+			line = fmt.Sprintf("  -%s, --%s", flag.Shorthand, flag.Name)
+		} else {
+			line = fmt.Sprintf("      --%s", flag.Name)
+		}
+
+		varname, usage := pflag.UnquoteUsage(flag)
+		if varname != "" {
+			line += " " + varname
+		}
+		if flag.NoOptDefVal != "" {
+			switch flag.Value.Type() {
+			case "string":
+				line += fmt.Sprintf("[=\"%s\"]", flag.NoOptDefVal)
+			case "bool":
+				if flag.NoOptDefVal != "true" {
+					line += fmt.Sprintf("[=%s]", flag.NoOptDefVal)
+				}
+			case "count":
+				if flag.NoOptDefVal != "+1" {
+					line += fmt.Sprintf("[=%s]", flag.NoOptDefVal)
+				}
+			default:
+				line += fmt.Sprintf("[=%s]", flag.NoOptDefVal)
+			}
+		}
+
+		// This special character will be replaced with spacing once the
+		// correct alignment is calculated
+		line += "\x00"
+		if len(line) > maxlen {
+			maxlen = len(line)
+		}
+
+		line += usage
+		if !defaultIsZeroValue(flag) {
+			if flag.Value.Type() == "string" {
+				line += fmt.Sprintf(" (default %q)", flag.DefValue)
+			} else {
+				line += fmt.Sprintf(" (default %s)", flag.DefValue)
+			}
+		}
+		if len(flag.Deprecated) != 0 {
+			line += fmt.Sprintf(" (DEPRECATED: %s)", flag.Deprecated)
+		}
+
+		lines = append(lines, line)
+	})
+
+	for _, line := range lines {
+		sidx := strings.Index(line, "\x00")
+		spacing := strings.Repeat(" ", maxlen-sidx)
+		// maxlen + 2 comes from + 1 for the \x00 and + 1 for the (deliberate) off-by-one in maxlen-sidx
+		fmt.Fprintln(buf, line[:sidx], spacing, wrap(maxlen+2, 0, line[sidx+1:]))
+	}
+
+	return buf.String()
+}
+
+// Splits the string `s` on whitespace into an initial substring up to
+// `i` runes in length and the remainder. Will go `slop` over `i` if
+// that encompasses the entire string (which allows the caller to
+// avoid short orphan words on the final line).
+func wrapN(i, slop int, s string) (string, string) {
```

Comment:
I "re-implemented" all dependency functions of `FlagUsagesWrapped` here to create our own version of it